src/models/surveymodel.h

```diff
@@ -33,6 +33,7 @@ class SurveyModel final : public QObject {
 
   Q_INVOKABLE void openCurrentSurvey();
   Q_INVOKABLE void dismissCurrentSurvey();
+  Q_INVOKABLE QString getCurrentSurveyId();
```

Comment:
make it a property:

`Q_PROPERTY(QString currentSurveyId MEMBER m_currentSurveyId NOTIFY hasSurveyChanged)`

And remove all the rest. This should be enough to have `VPNSurveyModel.currentSurveyId`